lib/ts/types.ts

```diff
@@ -125,6 +125,11 @@ export type SuperTokensPlugin = {
     id: string; // TODO: validate that no two plugins have the same id
     version?: string;
     compatibleSDKVersions?: string | string[]; // match the syntax of the engines field in package.json
+    init?: (
+        config: Omit<TypeInput, "recipeList" | "experimental">,
+        allPlugins: Pick<SuperTokensPlugin, "id" | "version" | "compatibleSDKVersions" | "exports">[],
+        sdkVersion: string
+    ) => Promise<void> | void;
```

Comment:
I don't see a usecase for an async initializer and we can't await it in the post-init hook anyway.